tests/unit/test_cli.py

```diff
@@ -81,13 +82,21 @@ class TestClientArguments(unittest.TestCase):
     Tests the client cli can parse all arguments it is supposed to
     and can initialize the runner in preparation for a request
     """
+    class ParseFailureException(Exception):
+        pass
+
+    def _raiseParseFailureException(self, exitVal):
+        raise self.ParseFailureException(exitVal)
+
     def setUp(self):
         self.parser = cli.getClientParser()
 
-    # TODO we need a way to test parse failures. This is tricky because
-    # argparse calls sys.exit() on error, which we can't catch directly as
-    # an exception. Using mock to intercept this call would at least
-    # verify that an error has been raised.
+    def testParseFailure(self):
+        cliInput = "invalidCommand"
+        with utils.suppressOutput():
```

Comment:
And we can't continue beyond that line, since `parse_args` (reasonably) assumes internally that the lines after a `sys.exit`do not execute, so if we make `sys.exit` a no-op, we get a different exception on the next line or whatever